### PR TITLE
Release 3.11.0

### PR DESCRIPTION
**Pull request by joaander**
## Description

Changes needed to release HOOMD-blue 3.11.0.

## Motivation and context

Make the recent changes available to users.

See checklist in #1536.

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -12,7 +12,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - name: Run stale action
-      uses: actions/stale@v7.0.0
```
Comment (b-butler) [inline .github/workflows/stale.yml:15]: This changes the name and version number. Given we use `actions/checkout` elsewhere at a version `3.5.2` this seems wrong.

**Review by b-butler — CHANGES_REQUESTED** [review]
There seems to be an inconsistency in the action update.

File: .github/workflows/stale.yml (hunk 1 above)
Comment (joaander) [reply .github/workflows/stale.yml:15]: Good catch. Fixed.

**Review by b-butler — APPROVED** [review]
:+1:

**Pull request merged**